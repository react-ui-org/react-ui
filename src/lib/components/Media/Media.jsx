import PropTypes from 'prop-types';
import React from 'react';
import { withProviderContext } from '../../provider';
import styles from './Media.scss';

export const Media = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.media}>
      {children}
    </div>
  );
};

Media.propTypes = {
  /**
   * Nested elements. Supported types are:
   * * `MediaBody`
   * * `MediaObject`
   */
  children: PropTypes.node.isRequired,
};

export const MediaWithContext = withProviderContext(Media, 'Media');

export default MediaWithContext;
