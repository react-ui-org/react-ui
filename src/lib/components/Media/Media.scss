@use "../../styles/tools/spacing";

.media {
    @include spacing.bottom(layouts);

    display: flex;
    align-items: flex-start;
}

.object {
    margin-right: spacing.of(3);
}

.body {
    flex: 1;
}
