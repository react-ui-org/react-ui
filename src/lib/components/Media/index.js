export { default as Media } from './Media';
export { default as MediaObject } from './MediaObject';
export { default as MediaBody } from './MediaBody';
