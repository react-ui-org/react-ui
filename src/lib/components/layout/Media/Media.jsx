import PropTypes from 'prop-types';
import React from 'react';
import styles from './Media.scss';

const Media = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.media}>
      {children}
    </div>
  );
};

Media.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Media;
