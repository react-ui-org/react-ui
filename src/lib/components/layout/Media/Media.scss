@use '../../../styles/settings/layouts';
@use '../../../styles/tools/spacing';

.media {
  display: flex;
  align-items: flex-start;
  margin-bottom: layouts.$common-bottom-spacing;
}

.object {
  margin-right: spacing.of(3);
}

.body {
  flex: 1;
}
