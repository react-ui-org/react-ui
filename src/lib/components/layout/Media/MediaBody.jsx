import PropTypes from 'prop-types';
import React from 'react';
import styles from './Media.scss';

const MediaBody = ({ children }) => (
  <div className={styles.body}>
    {children}
  </div>
);

MediaBody.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MediaBody;
