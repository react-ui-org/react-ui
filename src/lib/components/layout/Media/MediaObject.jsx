import PropTypes from 'prop-types';
import React from 'react';
import styles from './Media.scss';

const MediaObject = ({ children }) => (
  <div className={styles.object}>
    {children}
  </div>
);

MediaObject.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MediaObject;
