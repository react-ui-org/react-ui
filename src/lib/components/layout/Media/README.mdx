---
name: Media
menu: 'Components: Layouts'
route: /components/layout/media
---

# Media

import {
  Playground,
  Props,
} from 'docz'
import { Placeholder } from '../../../../docs/_components/Placeholder/Placeholder'
import { Media } from './Media'
import { MediaBody } from './MediaBody'
import { MediaObject } from './MediaObject'

The Media layout nicely aligns text content next to an object.

## Basic Usage

To implement the Media component, you need to import it first:

```js
import { Media, MediaBody, MediaObject } from '@react-ui-org/react-ui';
```

And use it:

<Playground>
  <Media>
    <MediaObject>
      <Placeholder bordered>Media object</Placeholder>
    </MediaObject>
    <MediaBody>
      <Placeholder bordered>Media body</Placeholder>
    </MediaBody>
  </Media>
</Playground>

See [API](#api) for all available options.

## General Guidelines

Use the Media layout for **highly repetitive layouts** like comments, article
teasers, etc., ie. when you need a **fixed-sized object** (eg. an image) and an
**area that expands** over the remaining horizontal space (eg. a paragraph).

## API

<Props of={Media} />

### MediaBody

A place for the text content.

<Props of={MediaBody} />

### MediaObject

A place for the media object.

<Props of={MediaObject} />

---

Next: [Toolbar â†’](/components/layout/toolbar)
