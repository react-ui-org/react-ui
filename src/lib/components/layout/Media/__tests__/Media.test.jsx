import React from 'react';
import { shallow } from 'enzyme';
import { Media } from '../Media';
import { MediaBody } from '../MediaBody';
import { MediaObject } from '../MediaObject';

describe('rendering', () => {
  it('renders correctly with a single child', () => {
    const tree = shallow((
      <Media>
        <MediaBody>body</MediaBody>
      </Media>
    ));

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple children', () => {
    const tree = shallow((
      <Media>
        <MediaObject>object</MediaObject>
        <MediaBody>body</MediaBody>
      </Media>
    ));

    expect(tree).toMatchSnapshot();
  });
});
