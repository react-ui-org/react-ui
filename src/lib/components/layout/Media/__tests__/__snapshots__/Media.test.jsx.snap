// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with a single child 1`] = `
<div
  className="media"
>
  <MediaBody>
    body
  </MediaBody>
</div>
`;

exports[`rendering renders correctly with multiple children 1`] = `
<div
  className="media"
>
  <MediaObject>
    object
  </MediaObject>
  <MediaBody>
    body
  </MediaBody>
</div>
`;
